flocker/node/agents/ebs.py

```diff
@@ -195,11 +317,74 @@ def resize_volume(self, blockdevice_id, size):
         pass
 
     # cloud_instance_id here too
-    def attach_volume(self, blockdevice_id, host):
-        pass
+    def attach_volume(self, blockdevice_id, attach_to):
+        """
+        Attach an EBS volume to given compute instance.
+
+        :param unicode blockdevice_id: EBS UUID for volume to be attached.
+        :param unicode attach_to: Instance id of AWS Compute instance to
+            attached the blockdevice to.
+
+        :raises UnknownVolume: If there does not exist a BlockDeviceVolume
+            corresponding to the input blockdevice_id.
+        :raises AlreadyAttachedVolume: If the input volume is already attached
+            to a device.
+        """
+        volume = self._get(blockdevice_id)
+        if volume.attached_to is not None:
+            raise AlreadyAttachedVolume(blockdevice_id)
+
+        with self.lock:
+            # begin lock scope
+
+            blockdevices = FilePath(b"/sys/block").children()
+            device = self._next_device(attach_to)
```

Comment:
Related to the comment in `_next_device`, the current failure mode for "no more devices can be attached" seems to be that `None` is returned and then passed in to the `self.connection.attach_volume` call (one line below this) which I'm guessing is an error because the AWS API docs say the device is a required argument?

Since the interface doesn't specify what to do in this error case I don't think anything needs to be fixed here but maybe this is a good place to link to the issue about making the interface specify what to do.